spec/lib/oauth/refresh_token_request_spec.rb

```diff
@@ -67,6 +69,53 @@ module Doorkeeper::OAuth
       expect(subject).to be_valid
     end
 
+    context 'refresh tokens expire on access token use' do
+      let(:server) do
+        double :server,
+               access_token_expires_in: 2.minutes,
+               custom_access_token_expires_in: ->(_oauth_client) { 1234 },
+               refresh_token_revoked_in: 1.day
+      end
+
+      it 'issues a new token for the client' do
+        expect do
+          subject.authorize
+        end.to change { client.access_tokens.count }.by(1)
+      end
+
+      it 'does not revoke the previous token' do
+        subject.authorize
+        refresh_token.revoked?.should eq(false)
+      end
+
+      it 'sets the previous refresh token in the new access token' do
+        subject.authorize
+        expect(client.access_tokens.last.previous_refresh_token).to eq(refresh_token.refresh_token)
+      end
+    end
+
+    context 'longer lived refresh tokens' do
+      let(:server) do
+        double :server,
+               access_token_expires_in: 2.minutes,
+               custom_access_token_expires_in: ->(_oauth_client) { 1234 },
+               refresh_token_revoked_in: 1.day
+      end
+
+      it 'revokes the previous token' do
+        expect { subject.authorize } .to change { refresh_token.revoked_at }.from(nil).to(be_within(1).of(DateTime.now + 1.day))
```

Comment:
> Do you mind if I leave these for #730

If Hound is complaining about code that looks like code elsewhere in the project, I think we should delete/ignore its comments and address them in other PRs indeed, if we find them better.